bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/AudioManager.java

```diff
@@ -142,6 +142,13 @@
     @Nullable
     AudioSource getSource();
 
+    /**
+     * Retrieves all audio sources
+     *
+     * @return all audio sources
+     */
+    Set<AudioSource> getAllSources();
```

Comment:
Adding methods to this interface makes this PR API breaking. I'm not aware of other AudioManager implementations but I think its not justified by this change.
There are two alternatives:
* Add both methods with a default implementation (returning empty lists) here.
* Add a new `AudioServiceProvider` service which gets both lists injected (like `AudioManagerImpl` does it right now) and use the new service in `AudioManagerImpl` and `AudioConsoleCommandExtension`.

Opinions @maggu2810, @kaikreuzer, @sjka?

Reply:
There must be only a single `AudioManager` service available in the system and it is the framework that provides the implementation, i.e. this interface is NOT meant as an SPI to implement, but as an API to use - and thus it imho isn't breaking and all fine for me.

Reply:
My only comment would be that this method is not absolutely required as the ids can already be retrieved and the sources can then be requested one by one, so that should work as well without touching the interface.

Reply:
You can only obtain the Source IDs and not the AudioSource objects with the `getSourceIds` methods. Without the AudioSource objects `AudioConsoleCommandExtension` is unable to show the labels.

Your comment would apply to the `getAllSinks()` method. It's does make the interface more more consistent and easier to use when it provides `getAll...()` methods for both sinks and sources.

Reply:
Ah, I missed that there is no method `getSource(String id)` - so yes, you are right that the method `getAllSources()` is indeed required.

Reply:
We could probably actually remove `getSourceIds` and `getSinkIds`, since those two methods are so far only used by the console to print the list - the do not have any other purpose as it seems.

Reply:
Yes but I think that would be more APIBreaking than adding the new methods. ;-)

Reply:
Well, it would at least prove @htreu right then and he can add the label again 😆

Reply:
The `getSinkIds` method is also used by the MediaActionTypeProvider:

https://github.com/eclipse/smarthome/blob/c77dbf832be9e8a532a461662ac53f0306371ed9/bundles/automation/org.eclipse.smarthome.automation.module.media/src/main/java/org/eclipse/smarthome/automation/module/media/internal/MediaActionTypeProvider.java#L133-L136

Furthermore, the `getSinks(String pattern)` method returns Sink IDs so a better name would have been `getSinkIds` which is also more consistent with the `getSourceIds` method.

Maybe it's better to rework this all in a separate PR?

Reply:
As you like!